src/mame/nmk/ddealer.cpp

```diff
@@ -186,13 +112,68 @@ class ddealer_state : public driver_device
 	// splitted foreground area
 	tilemap_t  *m_fg_tilemap_left;
 	tilemap_t  *m_fg_tilemap_right;
-
-	// MCU sim related
-	int   m_respcount;
-	u8    m_input_pressed;
-	u16   m_coin_input;
+	
+	u8 m_bus_status;
 };
 
+void ddealer_state::machine_start()
+{
+	save_item(NAME(m_bus_status));
+}
+
+void ddealer_state::machine_reset()
+{
+	m_bus_status = 0x04;
+}
+
+void ddealer_state::mcu_port6_w(u8 data)
+{
+	// the actual mechanism is a little more complex, but these are written at the
+	// start and end of the take / release bus function
```

Comment:
Can you elaborate a bit? Connected to bus request I presume?

Reply:
the notes I was sent

writes 0x03:  0011 (important thing is bit 3 is zero, asking for bus request)
reads bit 2, until it's zero (receiving bus grant signal from 68k)
writes 0x01:  0001  (important thing here is bit 1 is zero, sending bus grant acknowledge signal to 68k)
writes 0x08: 1000  (important thing here is bit 0 is zero, enabling the memory access logic in the pcb)
wirites 0x09: 1001 (important thing here is bit 0 is one, disabling the memory access logic in the pcb)

Reply:
@angelosa yes, it's 68k bus arbitration lines. Device is hooked to the main CPU in this way:

![image](https://github.com/mamedev/mame/assets/6969728/01e7c3d5-12be-4360-9a7d-3eed0ab571ed)

- Bus request line is bit 3 of Port 6 and configured as output
- Bus grant line is bit 2 of the Port 6 and it's configured as input
- Bus grant acknowledge line is bit 1 of the Port 6 and it's configured as output
- Also there is another pin that enables the pcb logic to access the bus after all of above it's done, that is the bit 0 of the Port 6, configured as output, too